lib/trento/infrastructure/checks/checks.ex

```diff
@@ -93,6 +94,18 @@ defmodule Trento.Infrastructure.Checks do
     commanded().dispatch(command, correlation_id: execution_id)
   end
 
+  defp build_env(%AscsErsClusterExecutionEnv{
```

Comment:
I am not sure we need a new `AscsErsClusterExecutionEnv` struct.
We could simplify by patter matching on the cluster type of a `ClusterExecutionEnv`.

- this function would get a `ClusterExecutionEnv` and it would match only if `cluster_type` is `:ascs_ers` (or using the enum `Trento.Clusters.Enums.ClusterType.ascs_ers()`)
- the following `build_env` would need to be refactored to match only when `cluster_type` is either `:hana_scale_up` or `:hana_scale_out`
- maybe an extra `build_env` function might be needed to handle non matching cluster type, but I am not sure. Anyway if needed should not be a big deal adding it

I was also thinking about the possibility to add an extra test for the change in this module, but I see it is already covered in the tests added in `clusters_test.ex` :+1:

Reply:
I didn't like the env being created with `nil` fields due to Elixir's structs behavior but it's ok, we can keep the code simpler.

Reply:
(BTW no extra function needed for non matching cluster type: when a clause doesn't match, the subsequent one gets matched against the payload. In the end if nothing matches the BEAM throws the error)